.github/workflows/build.yml

```diff
@@ -0,0 +1,79 @@
+name: Build Anmol
+
+on:
+  push:
+    branches: [ master, ibtida ]
+  workflow_dispatch:
+
+jobs:
+  build-anmol:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set testnet env
+        if: endsWith(github.ref, '/ibtida')
+        run: |
+          echo "ENVIRONMENT=testnet" >> $GITHUB_ENV
+
+      - name: Set production env
+        if: endsWith(github.ref, '/master')
+        run: |
+          echo "ENVIRONMENT=production" >> $GITHUB_ENV
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: |
+            /tmp/.buildx-cache-base
+            /tmp/.buildx-cache-prod
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
```

Comment:
I'm not quite sure about this caching. I saw this implementation recommended in an article, and the same is used by another chain doing a similar thing. However, things still seem slow to me despite a cache hit, which you can see in a [recent run ](https://github.com/anmolnetwork/anmol-node/runs/3277810744?check_suite_focus=true). It might be something to continue looking into as we go on.